src/deq_eid/main.py

```diff
@@ -0,0 +1,362 @@
+            config.MEDIA_FIELDS,
+            None,
+        )
+        records.extract_data_from_salesforce()
+
+        return records.df
+
+    def _publish_dataset(self, table_name, title, fields, sdf, type):
+        import arcpy
+
+        #: save to a feature class just so that we can add field aliases
+        self.skid_logger.info("saving to feature class...")
+        feature_class_path = self.tempdir_path / f"{table_name}.gdb" / table_name
+        if arcpy.Exists(feature_class_path):
+            self.skid_logger.info("deleting existing feature class...")
+            arcpy.management.Delete(feature_class_path)
+        elif not arcpy.Exists(feature_class_path.parent):
+            self.skid_logger.info("creating gdb...")
+            arcpy.management.CreateFileGDB(
+                str(feature_class_path.parent.parent),
+                feature_class_path.parent.name,
+            )
+
+        if type == "layer":
+            self.skid_logger.info("exporting to feature class...")
+            sdf.spatial.to_featureclass(
+                location=feature_class_path,
+                sanitize_columns=False,
+            )
+        else:
+            self.skid_logger.info("exporting to table...")
+            csv_file_path = self.tempdir_path / f"{table_name}.csv"
+            sdf.to_csv(csv_file_path, index=False)
+            arcpy.conversion.ExportTable(
+                str(csv_file_path),
+                str(feature_class_path),
+            )
+
+        self.skid_logger.info("adding field aliases...")
+        aliases = {field.agol_field: field.alias for field in fields}
+        for field in arcpy.Describe(str(feature_class_path)).fields:
+            if field.name in aliases:
+                arcpy.management.AlterField(
+                    str(feature_class_path),
+                    field.name,
+                    new_field_alias=aliases[field.name],
+                )
+
+        self.skid_logger.info("zipping and publishing...")
+        zip_path = self.tempdir_path / f"{table_name}.zip"
+        helpers.zip_fgdb(feature_class_path.parent, zip_path)
+        fgdb_item = self.gis.content.add(
+            {
+                "type": "File Geodatabase",
+            },
+            data=str(zip_path),
+            folder="Interactive Map",
+        )
+        layer_item = fgdb_item.publish(
+            publish_parameters={
+                "name": table_name if self.secrets.IS_DEV is False else f"{table_name}_dev",
+                "layerInfo": {
+                    "capabilities": "Query",
+                },
+            },
+        )
+        manager = arcgis.features.FeatureLayerCollection.fromitem(layer_item).manager
+        manager.update_definition({"capabilities": "Query,Extract"})
+        layer_item.update({"title": title})
+        if not self.secrets.IS_DEV:
+            layer_item.sharing.sharing_level = SharingLevel.EVERYONE
+
+        print(f"feature layer published: {title} | {layer_item.id}")
```

Comment:
**question**: should this be a log entry instead of a print?

Reply:
It doesn't matter because that method is only run locally.